vrpn_Joylin.C

```diff
@@ -51,8 +51,16 @@ vrpn_Joylin::vrpn_Joylin(const char * name,
 
 vrpn_Joylin::~vrpn_Joylin()
 {
+  if (fd >= 0) {
+    close(fd);
+    fd = -1;
+  }
+  if (device != NULL) {
+    free(device);
```

Comment:
You can call `free(device)` even if `device == NULL`; `free()` will just be a no-op in that case. So you can remove the `if` statement and just unconditionally call `free(device); device = NULL;`.

Reply:
That used to be the case, but since at least C99 (and maybe earlier), you can safely call `free()` on `NULL` pointers. Certainly saves some boilerplate!